editoast/src/main.rs

```diff
@@ -714,6 +714,36 @@ async fn refresh_search_tables(
     Ok(())
 }
 
+async fn refresh_search_tables_per_infra(
+    db_pool: Data<DbPool>,
+    infra_id: i64,
+) -> Result<(), Box<dyn Error + Send + Sync>> {
+    let objects: Vec<String> = vec!["signal".into(), "operational_point".into(), "track".into()];
+    dbg!(objects.clone());
+    let mut conn = db_pool.get().await?;
+    for object in objects {
+        let Some(search_config) = SearchConfigFinder::find(&object) else {
+            eprintln!("❌ No search object found for {object}");
+            continue;
+        };
+        if !search_config.has_migration() {
+            eprintln!("❌ No migration defined for {object}");
+            continue;
+        }
```

Comment:
Because some search tables that are not related to infras (eg. projects, studies, maybe rolling stocks soon) might match but we don't want to update those.

Besides, iterating over a list when you already know which items you expect seems like a bad pattern to me (admittedly irrelevant in practice since there won't be thousands of search tables anytime soon 🙃).